src/main/scala/sigmastate/lang/SigmaBinder.scala

```diff
@@ -93,10 +94,10 @@ class SigmaBinder(env: ScriptEnv, builder: SigmaBuilder,
     case Block(Seq(), body) => Some(body)
 
     case block @ Block(binds, t) =>
-      val newBinds = for (Val(n, t, b) <- binds) yield {
-        if (env.contains(n)) error(s"Variable $n already defined ($n = ${env(n)}")
+      val newBinds = for (Val(n, t, b, Nullable(srcCtx)) <- binds) yield {
+        if (env.contains(n)) error(s"Variable $n already defined ($n = ${env(n)}", srcCtx)
         val b1 = eval(b, env)
-        mkVal(n, if (t != NoType) t else b1.tpe, b1)
+        mkVal(n, if (t != NoType) t else b1.tpe, b1, srcCtx)
```

Comment:
The alternative using dynamic variable is 
`currentSrcCtx.withValue(srcCtx) { mkVal(n, if (t != NoType) t else b1.tpe, b1) }`
Thus inside `mkVal` the value of `srcCtx` will be available.

Reply:
Cool trick! Done.